packages/cli/src/__tests__/fix.ts

```diff
@@ -70,6 +71,328 @@ test("set main and module field", async () => {
   `);
 });
 
+test("set exports field when opt-in", async () => {
+  let tmpPath = await testdir({
+    "package.json": JSON.stringify({
+      name: "package-exports",
+      main: "index.js",
+      module: "dist/package-exports.esm.js",
+      preconstruct: {
+        exports: {
+          envConditions: ["worker", "browser"],
+        },
+        ___experimentalFlags_WILL_CHANGE_IN_PATCH: {
+          exports: true,
+        },
+      },
+    }),
+    "src/index.js": "",
+  });
+
+  await fix(tmpPath);
+
+  expect(await getFiles(tmpPath, ["package.json"])).toMatchInlineSnapshot(`
+    ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ package.json ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
+    {
+      "name": "package-exports",
+      "main": "dist/package-exports.cjs.js",
+      "module": "dist/package-exports.esm.js",
+      "browser": {
+        "./dist/package-exports.esm.js": "./dist/package-exports.browser.esm.js"
+      },
+      "exports": {
+        ".": {
+          "module": {
+            "worker": "./dist/package-exports.worker.esm.js",
+            "browser": "./dist/package-exports.browser.esm.js",
```

Comment:
Hm, I don't think it's ideal to rely on the "caller"/bundler being configured with a module condition. Rollup always includes the module condition but I'm pretty sure that webpack doesn't if you override `resolve.conditionNames` (I didn't test it though). Usually used condition names probably should be configured indirectly through `target` but it is not unlikely that somebody configures it explicitly. In such a case it is likely that with `conditionNames: ['browser']` they wouldn't pick up our `exports.module.browser`.

It also seems that esbuild doesn't support `module` condition by default - I've looked into its code and I don't see it there. Its docs also doesn't mention it: https://esbuild.github.io/api/#how-conditions-work

Throwing new tools, like bun, into the mix - we can't assume that `module` is always supported.

Reply:
Is this a saying we should change something here or just a "this sucks"? Because I agree it sucks but I don't see a way to make it better except for going Node ESM only (I think Node ESM only is the only practical way to go, even though it of course sucks for other reasons. I'm imagining that Preconstruct will have a mode for what it does now and a Node ESM only mode)

Reply:
> Are you saying we should change something here or just "this sucks"? Because I agree it sucks but I don't see a way to make it better except for going Node ESM only

Hm, I was only commenting on the `module` condition used with env conditions here. I agree that for now, we shouldn't allow node loading our ESM files.

Are you afraid here that even when using a bundler we could risk introducing a dual-package hazard when the module condition is not present because they might bundle using "node strict rules" when the exports are present? And that's why you have chosen to only allow those env conditions in combination with the `module` condition?

Reply:
That sort of problem, yes, basically "the semantics of what all this stuff does is not well defined so do the thing that seems the least likely to break"

Reply:
A slight problem here is that it will start loading CJS for browser targets without the module condition. But I guess the only offender here is esbuild right now. Perhaps we should open an issue there about this.